### PR TITLE
trialwise and cropped test decoding rewritten to use new model API, EpochsDataset…

### PR DESCRIPTION
**Pull request by sliwy**
#97 
I've tried to rewrite the `test/acceptance_tests/test_trialwise_decoding.py`. It does not reproduce the same results as the old test. I couldn't make skorch give the same results (even if I took exactly the same net and made batches sampling the same) as the previous code. Below there is a comparison of results, it is probably not really significant as the dataset contains only 60 observations, so I don't know if we should care a lot.
image
Epochs on x-axis.
 @robintibor let me know if you agree that current flow does not have to reproduce previous results.

One more thing is that I wanted to use the same dataset as in the previous test, so I had to read raw files with mne, then create epochs and pass it to the fit method. To do this I created a small dataset class implementing sampling from `mne.Epochs` object. `mne.Epochs` object as it is the simplest way to integrate with any different dataset that is now not implemented in braindecode. Code of the class is below (If you think it may be helpful we can add it to braindecode Dataset):
```python
import numpy as np
from torch.utils.data import Dataset
class EpochsDataset(Dataset):
    def __init__(self, windows):
        self.windows = windows
        self.y = np.array(self.windows.events[:, -1])
        self.y = self.y - self.y.min()

    def __getitem__(self, index):
        X = self.windows.get_data(item=index)[0].astype('float32')[:, :, None]
        y = self.y[index]
        return X, y

    def __len__(self):
        return len(self.windows.events)
````
Next thing will be to change reading data to use braindecode datasets but for now I can't do this - I need to spend more time on it as I get some errors. When I confirm that it's not my fault I will let you know where the problem is.

### CODE REVIEW

**Comment by robintibor** [comment]
I will have a closer look later, but to answer your question: Yes, new code does not have to reproduce numbers of old code, that's fine. No need to make them reproduce exactly from my view here. Too many random generators used differently to somehow set to same state etc.

Iw as also wondering about EpochsDataset, probably would be worthwhile to also have some function (outside of test), that takes X and y, wraps into mne and produces this kind of EpochsDataset.. so that people using other non-mne loading code have a clear way to enter their data. Such functionality could then be moved outside test, directly into main source

**Comment by sliwy** [comment]
It should be also possible to not use mne directly and `EpochsDataset`, instead we can use `MOABBDataset` class but I have some issues with this class, so I need to spend more time on it.

**Comment by robintibor** [comment]
hm ya also true, this is maybe a bit different points, what we want to use here in the test and the second task of having a converter x,y -> windowsdataset... could be a separate PR also... for now also fine if you just change the expected values in the test so that tests pass

**Comment by sliwy** [comment]
I changed also `test_cropped_decoding.py` to use `EEGClassifier`. Below the same plots as for `test_trialwise_decoding.py`. @robintibor do you think that I should modify something more?

image

**Comment by robintibor** [comment]
so now this is also mergeable right?

**Comment by sliwy** [comment]
rebased and ready to merge I think

**Comment by robintibor** [comment]
great
